libraries/AP_BattMonitor/AP_BattMonitor_Analog.cpp

```diff
@@ -108,7 +108,7 @@ AP_BattMonitor_Analog::AP_BattMonitor_Analog(AP_BattMonitor &mon,
 void
 AP_BattMonitor_Analog::read()
 {
-    if ((AP_BattMonitor::Type)_params._type.get() != AP_BattMonitor::Type::ANALOG_CURRENT_ONLY) {
+    if (_state.type != AP_BattMonitor::Type::ANALOG_CURRENT_ONLY) {
```

Comment:
this is a change in behaviour, ditto below

Reply:
Yes, this is why I was bringing it to your attention.

We had some nasty bugs in the CAN battery backend because we looked at the configured type rather than the sort of backend we *actually* allocated.

In the `read()` method here we *must* only look at the allocated type, *never* the configured-via-parameters type.  This patch makes sure we do that.  Whatever problems this might introduce are probably benign in this case, but we don't want the pattern in the code.

Reply:
Ok, need to change the title then :)

Reply:
> Ok, need to change the title then :)

Fair call, done